arch/arm64/src/common/arm64_syscall.c

```diff
@@ -321,35 +315,18 @@ uint64_t *arm64_syscall(uint64_t *regs)
         break;
     }
 
-  if (*running_task != tcb)
-    {
-      tcb = current_task(cpu);
-
 #ifdef CONFIG_ARCH_ADDRENV
       /* Make sure that the address environment for the previously
        * running task is closed down gracefully (data caches dump,
        * MMU flushed) and set up the address environment for the new
        * thread at the head of the ready-to-run list.
        */
 
-      addrenv_switch(NULL);
+      if (SYS_restore_context == cmd || SYS_switch_context == cmd)
+        {
+          addrenv_switch(NULL);
```

Comment:
should we remove the extra spaces

Reply:
https://github.com/apache/nuttx/pull/14991